cluster/src/main/java/org/apache/iotdb/cluster/log/manage/FilePartitionedSnapshotLogManager.java

```diff
@@ -117,11 +117,22 @@ private void collectTsFilesAndFillTimeseriesSchemas() throws IOException {
     collectTsFiles();
 
     // 2.register the measurement
+    boolean slotExistsInPartition = true;
     for (Map.Entry<Integer, Collection<TimeseriesSchema>> entry : slotTimeseries.entrySet()) {
       int slotNum = entry.getKey();
-      FileSnapshot snapshot = slotSnapshots.computeIfAbsent(slotNum, s -> new FileSnapshot());
-      if (snapshot.getTimeseriesSchemas().isEmpty()) {
-        snapshot.setTimeseriesSchemas(entry.getValue());
+
+      if (dataGroupMember.getMetaGroupMember() != null) {
+        List<Integer> slots =
+            ((SlotPartitionTable) dataGroupMember.getMetaGroupMember().getPartitionTable())
+                .getNodeSlots(dataGroupMember.getHeader());
```

Comment:
Move those codes before the for loop, we do not need to get the slots for this raft group per loop, it's not efficient.